custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/model/plugins/PluginInfoProvider.java

```diff
@@ -0,0 +1,39 @@
+package io.jenkins.tools.warpackager.lib.model.plugins;
+
+import io.jenkins.tools.warpackager.lib.config.DependencyInfo;
+import io.jenkins.tools.warpackager.lib.impl.plugins.MavenPluginInfoProvider;
+import io.jenkins.tools.warpackager.lib.impl.plugins.UpdateCenterPluginInfoProvider;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+/**
+ * Plugin Information provider.
+ * The implementations can be used to query various data about plugins.
+ * @see MavenPluginInfoProvider
+ * @see UpdateCenterPluginInfoProvider
+ * @since TODO
```

Comment:
P2: Don't forget to punch in the version number once you've made your decision.

Reply:
That's why it is a TODO comment ;)